### PR TITLE
New: parse caption tags in examples into a separate property. Fixes #131

### LINKED ISSUE #131: Support example caption
According to [usejsdoc's example documentation](http://usejsdoc.org/tags-example.html), it's valid to caption an example tag with `<caption>...</caption>`

```
/**
 * Solves equations of the form a * x = b
 * @example <caption>Example usage of method1.</caption>
 * // returns 2
 * globalNS.method1(5, 10);
 * @returns {Number} Returns the value of x for the equation.
 */
```

### PR DESCRIPTION
**Pull request by tmcw**

### CODE REVIEW

File: lib/doctrine.js [hunk 1]
```diff
@@ -526,6 +526,17 @@
             return true;
         };
 
+        TagParser.prototype.parseCaption = function parseDescription() {
+            var description = trim(sliceSource(source, index, this._last));
+            if (description.match(/^\s*<caption>([\s\S]+?)<\/caption>(\s*[\n\r])([\s\S]+)$/i)) {
+                this._tag.caption = RegExp.$1;
```
Comment (nzakas) [inline lib/doctrine.js]: A bit too much regex magic for my taste. Can you do this as just locating the tags and using slice() or substring() to get this info?

Also, please don't use the dollar properties on RegExp, these are deprecated in JS and aren't clear on what they refer to.

**Comment by nzakas** [comment]
Commit message nit: the tail should be "(fixes #131)". Haven't gotten the commit message checked setup yet. :-/

**Comment by tmcw** [comment]
Switched the method to use indexOf & substring instead of regex, and updated the commit message.

**Comment by nzakas** [comment]
Thanks, looks good. I'm going to hold off on merging for now because this is a breaking change and I want to see if there are other changes that are breaking coming as well.

**Comment by tmcw** [comment]
In the meantime, could you roll a bugfix release with #136 and #132? I would love to fix the downstream bugs fixed in those merges.

**Comment by nzakas** [comment]
I'll see if I can do it this week. Releases are still pretty manual, so it takes a bit of coordination to get release notes and such together.

**Comment by nzakas** [comment]
Just pushed 0.7.1.

**Comment by tmcw** [comment]
@nzakas I don't have any breaking PRs planned and don't see any in the queue. Could you roll a new release?

**Comment by nzakas** [comment]
Are you looking for this PR to be released or something else? 0.7.2 was released not too long ago.

**Comment by tmcw** [comment]
@nzakas I'm looking for this PR to be released

**Comment by nzakas** [comment]
I'll see if I can do it this week

**Comment by nzakas** [comment]
Sorry for the delay, got pretty sick the past week. I just published 1.0.0 with this change.

**Comment by tmcw** [comment]
Thanks, and hope you're feeling better @nzakas

**Pull request merged**